js/app.js

```diff
@@ -60,10 +60,31 @@ const App = {
 			if ($li != null) {
 				const editingId = $li.dataset.id;
 				const $input = App.$.getTodo($li).$input;
-				const selection = [$input.selectionStart, $input.selectionEnd];
+				const selection = {
+					start: $input.selectionStart, end: $input.selectionEnd
+				};
 				return {
 					id: editingId,
-					restore() { return App.$.setEditing(editingId, ...selection); }
+					restore(saveEvent) {
+						if (saveEvent.detail != null && !saveEvent.detail.isEcho) {
+							// Check whether the active to-do has been spliced
+							for (let matchingExpr of [].concat(
+								saveEvent.detail.update['@update']
+									.find(todo => todo['@id'] === editingId)?.title
+							)) {
+								// If so, we need to adjust the caret and
+								// selection based on the updates made
+								const [index, deleteCount, content] = matchingExpr?.['@splice'] ?? [];
+								if (index < selection.start) {
+									selection.start += content.length - deleteCount;
+								}
+								if (index < selection.end) {
+									selection.end += content.length - deleteCount;
+								}
+							}
+						}
+						return App.$.setEditing(editingId, selection.start, selection.end);
+					}
```

Comment:
This new code adjusts the selection of the input field based in incoming splices, to ensure that the cursor does not hop around when typing. In a real-world app this would be the responsibility of a text editing component.